src/support/nonce/manager.js

```diff
@@ -84,40 +84,43 @@ class NonceManager {
       ]);
 
       // Retrieve stored nonces for the provided account.
-      let nonces = await this.store.get(address);
+      const storedNonces = await this.store.get(address);
 
       // Keep nonces which are not mined yet
       // and release nonces which values are below the account tx count (i.e. lowest possible value).
-      nonces = _.pickBy(nonces, (value, nonce) => {
+      const assignedNonces = _.pickBy(storedNonces, (value, nonce) => {
         if (nonce >= txCount) return true;
         logger.debug(`Account '${address}', nonce released: ${Number(nonce)}`);
         return false;
       });
 
-      // Get all known transactions by combining local cache with data from tx pool.
-      const knownTransactions = _.assign({}, nonces, pending, queued);
+      // Get all known transactions by combining stored nonces with data from tx pool.
+      const knownTransactions = _.assign({}, assignedNonces, pending, queued);
 
-      // Get all used nonces.
-      const usedNonces = _.keys(knownTransactions);
-      if (usedNonces.length) {
-        logger.debug(`Account '${address}', used nonces: ${usedNonces.join(', ')}`);
+      // Get all known nonces.
+      const knownNonces = _.keys(knownTransactions);
+      if (knownNonces.length) {
+        logger.debug(`Account '${address}', used nonces: ${knownNonces.join(', ')}`);
       }
 
-      // Calculate max used nonce.
-      const maxUsedNonce = usedNonces.reduce((a, b) => Math.max(a, b), txCount);
+      // Calculate max known nonce.
+      const maxKnownNonce = knownNonces.reduce((a, b) => Math.max(a, b), txCount);
 
-      // Go from current tx count value (i.e. lowest possible value) to max known nonce looking for the gaps.
-      let nextNonce = txCount;
-      while (nextNonce <= maxUsedNonce) {
+      let nextNonce;
+      for (
```

Comment:
Not sure if its more readable now. What do you think?

Reply:
Yes, it's way better